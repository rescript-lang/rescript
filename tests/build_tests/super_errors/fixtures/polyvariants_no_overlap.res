let a: [#Resize | #KeyDown] = #Resize
let b: [#Click] = a
