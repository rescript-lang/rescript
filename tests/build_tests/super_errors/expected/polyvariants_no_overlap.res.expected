
  [1;31mWe've found a bug for you![0m
  [36m/.../fixtures/polyvariants_no_overlap.res[0m:[2m2:19[0m

  1 [2mâ”‚[0m let a: [#Resize | #KeyDown] = #Resize
  [1;31m2[0m [2mâ”‚[0m let b: [#Click] = [1;31ma[0m
  3 [2mâ”‚[0m 

  This has type: [1;31m[#KeyDown | #Resize][0m
  But it's expected to have type: [1;33m[#Click][0m
  
  These polymorphic variants are incompatible - they share no common constructors.